### ISSUE TITLE
Error with Zend\ServiceManager from ZF3

### ISSUE THREAD

**Issue by vertx-one** [zendframework/zend-barcode#41]
Try to use Barcode separate from ZF. 
I use composer for install 
"zendframework/zend-barcode": "2.7.0",
"zendframework/zend-config": "^2.6.0"

By default composer install requirement zend service manger from ZF3 which is not BC with ZF2
After switch to "zendframework/zend-servicemanager": "^2.7.8" error has gone away

### Code to reproduce the issue

```php
<?php
namespace Project;

use \Zend\Barcode\Barcode;
use \Zend\Barcode\Object\Code25interleaved;

class BarcodeGenerator {
    protected $config = [];

    function __construct($config = null)
    {
        $arConf = array(
            'barcode'        => Code25interleaved::class,
            'barcodeParams'  => [
                'text' => $config['text'],
                'withBorder' => true,
                'barHeight' => 38,
                'factor' => 2,
                'font' => 5
            ],
            'renderer'       => 'image',
            'rendererParams' => [
                'imageType' => 'png'
            ],
        );

        $this->config = new \Zend\Config\Config($arConf);
    }

    public function draw(){
        $barcode = Barcode::factory($this->config);
        return $barcode->draw();
    }
    public function render(){
        $barcode = Barcode::factory($this->config);
        $barcode->render();
    }
}
```

### Expected results
Rendered barcode

### Actual results
Deprecated: Zend\ServiceManager\AbstractPluginManager::__construct now expects a Interop\Container\ContainerInterface instance representing the parent container; please update your code in ./vendor/zendframework/zend-servicemanager/src/AbstractPluginManager.php on line 85

**Comment by weierophinney** [comment]
I cannot reproduce the issue using the code you've provided above.

To recreate, I created a new unit test, `ZendTest\Barcode\BarcodeFactoryTest`, with a single test method, `testFactoryDoesNotProduceServiceManagerDeprecations()`. The test contained the default configuration you provided above, substituting some text for the "text" property; I also did not push it into a `Zend\Config\Config` instance, as `Barcode::factory()` converts such values to an array anyways.

I then did the following statements:

```php
$barcode = Barcode::factory($config);
$this->assertInstanceOf(\Zend\Barcode\Renderer\Image::class, $barcode);
```

This test passes against both zend-servicemanager 2.7.8 (lowest supported zend-servicemanager version), 3.3.0 (the version in the lock file), and 3.3.2 (latest release). By default, when PHPUnit encounters errors of any kind, including deprecations, it will re-cast these as exceptions, causing a test error. No such errors occurred.

Is it possible you're using an older version of zend-barcode than you reported here? All of these tests were run against latest master, which is 2.7.1-dev currently.
